protocol/messenger.go

```diff
@@ -2043,24 +2044,84 @@ func (m *Messenger) Mailservers() ([]string, error) {
 	return nil, ErrNotImplemented
 }
 
-func (m *Messenger) initChatFirstMessageTimestamp(chat *Chat) error {
-	if !chat.CommunityChat() || chat.FirstMessageTimestamp != FirstMessageTimestampUndefined {
-		return nil
+func (m *Messenger) initChatsFirstMessageTimestamp(communityCache map[string]*communities.Community, chats []*Chat) {
```

Comment:
```suggestion
func (m *Messenger) initChatsFirstMessageTimestamp(communityCache *map[string]*communities.Community, chats []*Chat) {
```
this way, you're passing the pointer/reference and you can change the original map and also it's lighter since it's not a new value

Reply:
are you sure we need the extra pointer?
```go
type communityInfo struct {
	ID string
}

func TestPassMap(t *testing.T) {
	communityInfoMap := make(map[string]*communityInfo)
	communityInfoMap["1"] = &communityInfo{}
	updateCommunityInfo(communityInfoMap)
	require.Equal(t, "666", communityInfoMap["2"].ID)
}

func updateCommunityInfo(communityInfoMap map[string]*communityInfo) {
	communityInfoMap["2"] = &communityInfo{ID: "666"}
}
```

Reply:
Ah, since the `communityInfo` is a pointer, it still works! All good then